pkg/objstore/azure/helpers.go

```diff
@@ -37,11 +38,27 @@ func init() {
 }
 
 func getAzureStorageCredentials(logger log.Logger, conf Config) (blob.Credential, error) {
-	if conf.MSIResource != "" {
-		msiConfig := auth.NewMSIConfig()
-		msiConfig.Resource = conf.MSIResource
+	if conf.MSIResource != "" || conf.UserAssignedID != "" {
+		resource := conf.MSIResource
+		if resource == "" {
+			resource = fmt.Sprintf("https://%s.%s", conf.StorageAccountName, conf.Endpoint)
+		}
+
+		var (
+			spt *adal.ServicePrincipalToken
+			err error
+		)
+
+		if conf.UserAssignedID != "" {
```

Comment:
What if there is a function like `getServicePrincipalToken` which just returns the `*adal.ServicePrincipalToken`? That removes those two vars defines and allows for nice returns.

Reply:
Nice suggestion